### PR TITLE
[ES-143] [ES-144] [ES-150] [ES-153] [ES-167] [ES-168] various bug fixes

### PR DESCRIPTION
**Pull request by zesu22**
- [ES-143] Cancel button while scanning devices to cancel the ongoing scanning
- [ES-144] Default port range issue
- [ES-150] RTL not working for `ara` 3 letter language code
- [ES-153] DiscTimeout issue while getting devices
- [ES-167] Partial error message on device not found
- [ES-168] capture_timeout & biometric_capture_failed error message properly handled and shown

[ES-143]: https://mosip.atlassian.net/browse/ES-143?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ES-144]: https://mosip.atlassian.net/browse/ES-144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ES-150]: https://mosip.atlassian.net/browse/ES-150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ES-153]: https://mosip.atlassian.net/browse/ES-153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ES-167]: https://mosip.atlassian.net/browse/ES-167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ES-168]: https://mosip.atlassian.net/browse/ES-168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: secure-biometric-interface-integrator/lib/standardConstant.js [hunk 1]
```diff
@@ -5,6 +5,13 @@ const states = {
   AUTHENTICATING: "AUTHENTICATING",
 };
 
+const ErrorCode = {
+  DEVICE_NOT_FOUND: "device_not_found_msg",
+  BIOMETRIC_CAPTURE_FAILED: "biometric_capture_failed_msg",
+  CAPTURE_TIMEOUT: "capture_timeout",
```
Comment (anshulv1401) [inline secure-biometric-interface-integrator/lib/standardConstant.js:11]: capture_timeout error code is not required

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js [hunk 2]
```diff
@@ -25,7 +26,9 @@ import {
   DeviceState,
   DeviceStateStatus,
   DEFAULT_PROPS,
+  ErrorCode,
 } from "./standardConstant";
+import { isAxiosError } from "axios";
```
Comment (anshulv1401) [inline secure-biometric-interface-integrator/lib/secureBiometricInterface.js:31]: IsAxiousError is not used, can be removed

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js [hunk 3]
```diff
@@ -114,22 +121,44 @@ class SecureBiometricInterface {
...
-   * @returns HTMLElement loader indicator
+   * Calling loading indicator
+   * @returns HTMLElement loading indicator with cancel button
    */
-  generateLoadingIndicator = (msg) => loadingIndicator(msg, this.isRtl);
+  generateLoadingIndicator = () => {
+    const elemArray = [
+      loadingIndicator(this.generateStatusMessage(), this.isRtl),
+    ];
+    if (this.status === states.LOADING) {
```
Comment (anshulv1401) [inline secure-biometric-interface-integrator/lib/secureBiometricInterface.js:131]: could add a state for DISCOVERING

File: secure-biometric-interface-integrator/lib/standardConstant.js (hunk 1 above)
Comment (zesu22) [reply secure-biometric-interface-integrator/lib/standardConstant.js]: fix in this commit [5ebb03c](https://github.com/mosip/esignet-plugins/pull/31/commits/5ebb03c7250691483cd36883d9cac477a1431126)

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js (hunk 2 above)
Comment (zesu22) [reply secure-biometric-interface-integrator/lib/secureBiometricInterface.js]: fix in this commit [5ebb03c](https://github.com/mosip/esignet-plugins/pull/31/commits/5ebb03c7250691483cd36883d9cac477a1431126)

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js (hunk 3 above)
Comment (zesu22) [reply secure-biometric-interface-integrator/lib/secureBiometricInterface.js]: fix in this commit [5ebb03c](https://github.com/mosip/esignet-plugins/pull/31/commits/5ebb03c7250691483cd36883d9cac477a1431126)

**Comment by zesu22** [comment]
image

Cancel button while component searching for biometric devices

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js [hunk 4]
```diff
@@ -548,14 +590,29 @@ class SecureBiometricInterface {
...
+  cancelLoadingIndicator() {
+    this.discoveryCancellationFlag = true;
+    this.errorStateChanged(
+      {
+        errorCode: ErrorCode.DEVICE_NOT_FOUND,
+        defaultMsg: "Device not found",
+      },
+      false
+    );
+    this.statusChanged(states.LOADED);
```
Comment (gsasikumar) [inline secure-biometric-interface-integrator/lib/secureBiometricInterface.js:605]: Why is the status changed to loaded?

**Review by gsasikumar — APPROVED** [review]

File: secure-biometric-interface-integrator/lib/secureBiometricInterface.js (hunk 4 above)
Comment (zesu22) [reply secure-biometric-interface-integrator/lib/secureBiometricInterface.js:605]: It actually denote that the component is currently not in a LOADING or AUTHENTICATING state, thats why we used LOADED

Comment (anshulv1401) [reply secure-biometric-interface-integrator/lib/secureBiometricInterface.js:605]: @gsasikumar , LOADED is a state representing no asynchronous call.

**Pull request merged**